sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -163,6 +163,51 @@ final class CastExprMeta[INPUT <: UnaryExpression with TimeZoneAwareExpression w
   override protected val needTimezoneTagging: Boolean = false
 }
 
+trait CastOptions extends Serializable {
+  // The brackets that are used in casting structs and maps to strings
+  def leftBracket: String
+
+  def rightBracket: String
+
+  // The string value to use to represent null elements in array/struct/map.
+  def nullString: String
+
+  def useDecimalPlainString: Boolean
+
+  def useHexFormatForBinary: Boolean
+
+  def stringToDateAnsiModeEnabled: Boolean
+
+  def ansiMode: Boolean
+
+  def legacyCastComplexTypesToString: Boolean
+}
+
+object DefaultCastOptions extends SparkConfCastOptions(false, false, false)
+
+object ArithmeticCastOptions {
+  def apply(failOnError: Boolean): SparkConfCastOptions =
+    new SparkConfCastOptions(false, failOnError, false)
```

Comment:
Why create a new instance when there are only two options?

```
object SparkConfCastOptions {
  val DEFAULT = SparkConfCastOptions(false, false, false)
  val ARITH_ANSI_OPTS =  SparkConfCastOptions(false, true, false)

  def arithmeticCast(failOnError: Boolean): SparkConfCastOptions = if (failOnError) {
    ARITH_ANSI_OPTS
  } else {
    DEFAULT
  }
}
```

Reply:
The third one will be ToPrettyStringOptions. 

But yeah, this should be a singleton